api/v1beta1/hypershiftagentserviceconfig_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HypershiftAgentServiceConfigSpec defines the desired state of HypershiftAgentServiceConfig.
+type HypershiftAgentServiceConfigSpec struct {
+	AgentServiceConfigSpec `json:",inline"`
+
+	// RemoteKubeconfigRef is a reference to the secret containing the kubeconfig for the destination Hypershift instance.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Hypershift kubeconfig reference"
+	RemoteKubeconfigRef corev1.LocalObjectReference `json:"remoteKubeconfigRef"`
```

Comment:
I'm not sure "Remote" is intuitive or valuable here. The hypershift control plane and this controller are running in the same namespace of the same cluster. They might even be on the same node, which is "closer" than the host cluster's api-server could be.

Reply:
Maybe just rollback to 'KubeconfigSecretRef' like named in hive?
https://github.com/openshift/assisted-service/blob/84b83af19531313df8186ea4f075894fb518f9a8/vendor/github.com/openshift/hive/apis/hive/v1/clusterrelocate_types.go#L9-L11

Reply:
Yes, that's what I would go with.